### ISSUE TITLE
deprecation warning regarding tarfile module when running tests with Python 3.12

### ISSUE THREAD

**Issue by FelixSchwarz** [borgbackup/borg#7716]
I saw this when running the test suite for the master branch using Python 3.12b3:

```
src/borg/testsuite/archiver/transfer_cmd.py::ArchiverTestCase::test_transfer_upgrade
  /usr/lib64/python3.12/tarfile.py:2213: DeprecationWarning: Python 3.14 will, by default, filter extracted tar archives and reject files or modify their metadata. Use the filter argument to control this behavior.
    warnings.warn(
```

I suspect it is a new warning but I'm not sure. Could be a good first issue for people starting with borg development. Fedora ships a Python 3.12 interpreter which can be used to reproduce the warning.

**Comment by ThomasWaldmann** [comment]
@FelixSchwarz 7976388868739b3eb84994ed93c028b4d5a5f8a0 see there.

I checked the CI test run with python 3.12 before that changeset was pushed, but didn't see a DeprecationWarning.

Also, no py312 on my local machine nor a Fedora VM at hand, so this is untested.

**Comment by FelixSchwarz** [comment]
I can confirm that the deprecation warning is gone when using the commits from PR #7592 , thank you for taking care of that.

**Comment by ThomasWaldmann** [comment]
PR #7592 merged, so this is fixed in master.

**Comment by ThomasWaldmann** [comment]
PR #7844 fixes this in 1.2-maint.
